specification/impact/Impact.Management/commonModels.tsp

```diff
@@ -0,0 +1,241 @@
+model Connectivity {
+  @doc("Protocol used for the connection")
+  protocol?: Protocol;
+
+  @doc("Port number for the connection")
+  port?: int32;
+
+  @doc("Source from which the connection was attempted")
+  source?: SourceOrTarget;
+
+  @doc("target which connection was attempted")
+  target?: SourceOrTarget;
+}
+
+@doc("Max and Min Threshold values for the metric")
+model ExpectedValueRange {
+  @doc("Min threshold value for the metric")
+  min: float64;
+
+  @doc("Max threshold value for the metric")
+  max: float64;
+}
+
+@doc("ARM error code and error message associated with the impact")
+model ErrorDetailProperties {
+  @doc("ARM Error code associated with the impact.")
+  errorCode?: string;
+
+  @doc("ARM Error Message associated with the impact")
+  errorMessage?: string;
+}
+
+@doc("Information about the impacted workload")
+model Workload {
+  @doc("the scenario for the workload")
+  context?: string;
+
+  @doc("Tool used to interact with Azure. SDK, AzPortal, etc.., Other")
+  toolset?: Toolset;
+}
+
+@doc("Client incident details ex: incidentId , incident source")
+model ClientIncidentDetails {
+  @doc("Client incident id. ex : id of the incident created to investigate and address the impact if any.")
+  clientIncidentId?: string;
+
+  @doc("Client incident source. ex : source system name where the incident is created")
+  clientIncidentSource?: IncidentSource;
+}
+
+@doc("List of incident interfaces.")
+union IncidentSource {
+  string,
+
+  /** When source of Incident is AzureDevops */
+  AzureDevops: "AzureDevops",
+
+  /** When source of Incident is Microsoft ICM */
+  ICM: "ICM",
+
+  /** When source of Incident is Jira */
+  Jira: "Jira",
+
+  /** When source of Incident is ServiceNow */
+  ServiceNow: "ServiceNow",
+
+  /** When source of Incident is Other */
+  Other: "Other",
+}
+
+@doc("List of impact categories.")
+union ImpactCategoryEnum {
+  string,
+
+  /** Availability category */
+  Availability: "Availability",
+
+  /** Performance category */
+  Performance: "Performance",
+
+  /** Other category */
+  Other: "Other",
+}
+
+@doc("List of azure interfaces.")
+union Toolset {
+  string,
+
+  /** If communication toolset is Terraform */
+  Terraform: "Terraform",
+
+  /** If communication toolset is Puppet */
+  Puppet: "Puppet",
+
+  /** If communication toolset is Chef */
+  Chef: "Chef",
+
+  /** If communication toolset is SDK */
+  SDK: "SDK",
+
+  /** If communication toolset is Ansible */
+  Ansible: "Ansible",
+
+  /** If communication toolset is ARM */
+  ARM: "ARM",
+
+  /** If communication toolset is Portal */
+  Portal: "Portal",
+
+  /** If communication toolset is Shell */
+  Shell: "Shell",
+
+  /** If communication toolset is Other */
+  Other: "Other",
+}
+
+#suppress "@azure-tools/typespec-azure-core/use-extensible-enum" "This really is a fixed enum"
```

Comment:
what is the benefit of calling it a fixed enum?

Reply:
we had this for these for our last 3 versions and in our current version, we are concerned that if we make this extensible, people might be sending the values that are not fixed as this will be modeled as string if they are not using SDK. we will consider making this extensible for our future versions